tests/test_timeseries_dataframe.py

```diff
@@ -128,6 +128,66 @@ def shape(self):
     def index(self):
         return self._index_data.ndarray
 
+    def sort(self, columns=list(), index_column=None, inplace=True):
+        """
+        Sort the dataframe along the given index column
+
+        :param columns: column names required in reordered TimeSeriesDataFrame
+        :type columns: List[str]
+        :param index_column: column name treated as the index, if None is
+                                given, sort along the index
+        :type index_column: Option[str]
+        :param inplace: flag indicates whether to sort inplace or out-of-place
+        :type inplace: bool
+        :return: sorted TimeSeriesDataFrame (return self if inplace is set to
+                 True)
+        """
+
+        if index_column is None and self._index_data is None:
+            raise Exception("TimeSeriesDataFrame: data frame has no index")
+
+        index_data = self._index_data if (
+            index_column is None
+            ) else self._data[self._columns.index(index_column)]
+        sorted_indices = index_data.argsort()
+
+        if inplace:
+            for i, col in enumerate(self._data):
+                self._data[i] = col.take_along_axis(sorted_indices)
+
+            if self._index_data is not None:
+                self._index_data = self._index_data.take_along_axis(
+                    sorted_indices)
+
+            return self
+        else:
+            ret = TimeSeriesDataFrame()
+            ret._index_name = self._index_name
+            ret._index_data = self._index_data.take_along_axis(sorted_indices)
+
+            for name in columns:
+                if name not in self._columns:
+                    raise Exception("Column '{}' does not exist".format(name))
+                new_col = self._data[self._columns.index(name)
```

Comment:
You can break the line in a nicer way:

```python
_ = self._columns.index(name)
new_col = self._data[_].take_along_axis(sorted_indices)
```

`_` is a idiomatic name for temporary surrogate for line-breaking.

Alternately, you can give the temporary variable a meaningful name:

```python
i = self._columns.index(name)
new_col = self._data[i].take_along_axis(sorted_indices)
```